README.md

```diff
@@ -1,13 +1,104 @@
-erl_optics
-=====
+# erl_optics
 
-An OTP library
+An OTP application for [optics](https://github.com/RAttab/optics).
 
-### Requirements
+## Requirements
+-----
 
 * Erlang OTP >= 21
+* [optics](https://github.com/RAttab/optics) v1.1.9
 
-Build
+## Building
 -----
-
     $ rebar3 compile
+
+## Usage
+-----
+
+For now, the only working mode is Carbon.
+
+need env :
+```
+  {erl_optics, [
+    {hostname, "<address of carbon host>"},
+    {port, <port number>},
+    {mode, carbon},
+    {interval, <polling interval time (ms)>}
+    ]}
+```
+
+To start the application:
+```
+  erl_optics_app:start(),
+```
+
+To start the optics polling:
+```
+  erl_optics:start_optics(<<"<namespace>">> [, <list of lenses>]),
+```
+The namespace will be prepending the name of every metric, for
+example: `"optics_namespace.my_counter"`
+
+The only lenses that require initial declaration are histo and quantile. If
+they are unused, they can be omitted, otherwise, they must be declared like so:
+
+```
+  Lenses = [erl_optics_lens:histo(<<"histo_name">>, [10, 20, 30, 40]),
+            erl_optics_lens:quantile(<<"quantile_name">>, 0.50, 10.0, 0.5)],
+  erl_optics:start_optics(<<"optics_namespace">>, Lenses).
+```
+
+## Lenses
+
+### counter
+A simple counter, polling will give the total increment value. Increment
+defaults to 1.
+```
+  erl_optics:counter_inc(<<"<counter_name>">> [, <increment>]),
+```
+
+### gauge
+Will keep the last value given to it.
+```
+  erl_optics:gauge_set(<<"<gauge_name>">>, <value>),
+```
+
+### dist
+Gives an estimation of the distribution of the values given to it. It provides
+metrics for 50th, 90th and 99th percentiles, the total number of values fed to
+it and the maximum value.
```

Comment:
I think the min is also provided

Reply:
Oh wow you're right. Welp, I think I'm making you the new maintainer of optics as you clearly know it better then me. Congratulations.